lib/ui/payment/payment_web_page.dart

```diff
@@ -0,0 +1,249 @@
+import 'dart:io';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_inappwebview/flutter_inappwebview.dart';
+import 'package:logging/logging.dart';
+import 'package:photos/models/subscription.dart';
+import 'package:photos/services/billing_service.dart';
+import 'package:photos/services/user_service.dart';
+import 'package:photos/utils/dialog_util.dart';
+import 'package:photos/utils/toast_util.dart';
+
+import '../loading_widget.dart';
+import '../progress_dialog.dart';
+
+const kWebPaymentRedirectUrl = "https://payments.ente.io/frameRedirect";
+const kWebPaymentBaseEndpoint = String.fromEnvironment("web-payment",
+    defaultValue: "https://payments.ente.io");
+
+class PaymentWebPage extends StatefulWidget {
+  final String planId;
+  final String actionType;
+
+  const PaymentWebPage({Key key, this.planId, this.actionType})
+      : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => _PaymentWebPage();
+}
+
+class _PaymentWebPage extends State<PaymentWebPage> {
+  final _logger = Logger("PaymentWebPage");
+  UserService userService = UserService.instance;
+  BillingService billingService = BillingService.instance;
+  ProgressDialog _dialog;
+  InAppWebViewController webView;
+  double progress = 0;
+  Uri initPaymentUrl;
+  String basePaymentUrl = kWebPaymentBaseEndpoint;
+
+  @override
+  void initState() {
+    userService.getPaymentToken().then((token) {
+      initPaymentUrl = _getPaymentUrl(token);
+      setState(() {
+      });
+    });
+    if (Platform.isAndroid && kDebugMode) {
+      AndroidInAppWebViewController.setWebContentsDebuggingEnabled(true);
+    }
+    _dialog = createProgressDialog(context, "please wait...");
+    super.initState();
+  }
+
+  Uri _getPaymentUrl(String paymentToken) {
```

Comment:
We usually try to order methods based on visibility such that public methods are defined first.